### PR TITLE
fix: Issue #1638 - replaced id_token by access_token for azure

### PR DESCRIPTION
**Pull request by melazarus**
### Description

This fixes the issue descript in #1638 where the upn can't be located in id_token but actually is present in access_token.

### ADDITIONAL INFORMATION
- [x] Has associated issue: 1638
- [x] Is Auth, RBAC security related.

### CODE REVIEW

File: flask_appbuilder/security/manager.py [hunk 1]
```diff
@@ -595,7 +595,7 @@ def get_oauth_user_info(self, provider, resp):
         # active-directory-protocols-oauth-code
         if provider == "azure":
             log.debug("Azure response received : {0}".format(resp))
-            id_token = resp["id_token"]
+            id_token = resp["access_token"]
```
[the remark this answers is not in the document]
Comment (melazarus) [reply flask_appbuilder/security/manager.py]: Thanks for reviewing, I have renamed the var now.

**Comment by thesuperzapper** [comment]
@melazarus can you clarify if something has changed on the azure end, and why we would have previously used `id_token`?

If possibly can you link me the relevant Azure docs?

**Comment by melazarus** [comment]
@thesuperzapper, as far as I can see, but I'm not an expert, this did not change in azure.
When I look at the documentation for [id_token](https://docs.microsoft.com/en-us/azure/active-directory/develop/id-tokens) and [access_token](https://docs.microsoft.com/en-us/azure/active-directory/develop/access-tokens) on the ms docs pages I see that the UPN is included in the payload of the access_token but it is not in the claims of the id_token.

**Comment by dpgaspar** [comment]
@melazarus @thesuperzapper 

Yes, this makes sense according to docs, this new feature was added here: https://github.com/dpgaspar/Flask-AppBuilder/pull/615

Looking at the docs:
The unique fields `email` and `username` use `upn` and `oid` respectively. Where in fact `upn` is not present on the `id_token` but `email` is, so it probably makes more sense to use this field (config would need to change to include the email option claim), and a fall back to construct a unique email based on the username has a best effort and to preserve uniqueness.

On the other hand `given_name` and `family_name` belong to the `access_token` claims. 

Not an expert on Azure AD, but `id_token` is more compliant to OpenID OAuth and `access_token` seems more Azure proprietary is this a right assumption @melazarus ?

**Comment by haripraghash** [comment]
Any details on why this PR was closed? `upn` issue still exists

**Comment by thesuperzapper** [comment]
> Any details on why this PR was closed? `upn` issue still exists

@haripraghash this was closed by a stalebot (not a human) due to inactivity.
Can you explain what you mean by `upn` issue?

**Comment by haripraghash** [comment]
@thesuperzapper Thank you for that.

I am referring to the issue #1638 . In that, it mentions that `upn` is not present in `id_token` but in `access_token`

**Pull request closed without merging**